### PR TITLE
Feature/#104 social

### PR DESCRIPTION
**Pull request by Zeavee**

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/FriendsFragment.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                recyclerView.setLayoutManager(new GridLayoutManager(context, mColumnCount));
+            }
+            //recyclerView.setAdapter(new MyFriendsRecyclerViewAdapter(new ArrayList<>(), mListener));
+        }
+        return view;
+    }
+
+
+    @Override
+    public void onAttach(Context context) {
```
[the remark this answers is not in the document]
Comment (Zeavee) [reply app/src/androidTest/java/ch/epfl/sdp/FriendsFragment.java]: It's an Override method, there already is a javadoc

File: app/src/androidTest/java/ch/epfl/sdp/FriendsFragment.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        if (context instanceof OnListFragmentInteractionListener) {
+            mListener = (OnListFragmentInteractionListener) context;
+        } else {
+            throw new RuntimeException(context.toString()
+                    + " must implement OnListFragmentInteractionListener");
+        }
+    }
+
+    @Override
+    public void onDetach() {
```
[the remark this answers is not in the document]
Comment (Zeavee) [reply app/src/androidTest/java/ch/epfl/sdp/FriendsFragment.java]: Same as above :)

File: app/src/main/java/ch/epfl/sdp/artificial_intelligence/GenPoint.java [hunk 3]
```diff
@@ -1,23 +1,22 @@
 package ch.epfl.sdp.artificial_intelligence;
 
 public abstract class GenPoint {
-    float arg1;
-    float arg2;
+    double arg1;
```
[the remark this answers is not in the document]
Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/artificial_intelligence/GenPoint.java]: Those are not my changes, they are only weird double branches things

File: app/src/main/java/ch/epfl/sdp/artificial_intelligence/PolarPoint.java [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
 public class PolarPoint extends GenPoint{
-    public PolarPoint(float r, float theta) {
+    public PolarPoint(double r, double theta) {
         super(r, theta);
     }
 
     @Override
     public CartesianPoint toCartesian() {
-        return new CartesianPoint(Double.valueOf(arg1*Math.cos(arg2)).floatValue(),Double.valueOf(arg1*Math.sin(arg2)).floatValue());
+        return new CartesianPoint(arg1 * Math.cos(arg2), arg1 * Math.sin(arg2));
```
[the remark this answers is not in the document]
Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/artificial_intelligence/PolarPoint.java]: Not in PolarPoint

File: app/src/main/java/ch/epfl/sdp/artificial_intelligence/PointConverter.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import ch.epfl.sdp.map.GeoPoint;
+import de.jotschi.geoconvert.GeoConvert;
+
+public class PointConverter {
+    public static GenPoint GeoPointToGenPoint(GeoPoint geoPoint) throws Exception {
+        double[] xy = GeoConvert.toUtm(geoPoint.longitude(), geoPoint.latitude());
+        return new CartesianPoint(xy[0], xy[1]);
+    }
+
+    public static GeoPoint GenPointToGeoPoint(GenPoint genPoint, GeoPoint refGeoPoint) {
```
[the remark this answers is not in the document]
Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/artificial_intelligence/PointConverter.java]: We should yes

File: app/src/main/java/ch/epfl/sdp/artificial_intelligence/PolarPoint.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/artificial_intelligence/PolarPoint.java]: But we could use r and phi (or theta)

[the remark this answers is not in the document]
Comment (varathar53) [reply app/src/main/java/ch/epfl/sdp/artificial_intelligence/PolarPoint.java]: perhabs to change it to r and phi/theta because its math stuff?

**Pull request merged**